### ISSUE TITLE
Use https to reach kalliope API

### ISSUE THREAD

**Issue by JulienSambre** [kalliope-project/kalliope-app#27]
Hi!

I'm currently using the app for sending orders to kalliope.

There are some anoying bugs (like text of orders appearing in chat text box, etc), but i can deal with them, and all in all it works.

However, i can't reach the Kalliope API on https and i had to open port 80 on my proxy and let the app communicate on http, which is highly unsecure.

Am I missing something and if not would it be possible de call an https endpoint?

Thanks!

**Comment by Sispheor** [comment]
It's not the job of kalliope to implement the ssl termination. Add a reverse proxy in front for that.

**Comment by JulienSambre** [comment]
I have done that, but if I put in URL field one of the following, none of them works :

subdomain.domain.name:443
https://subdomain.domain.name:443
https://subdomain.domain.name

I have the error : "Connection failed : Response with status: 0 for URL: null". I have no problem if I put : "subdomain.domain.name:80" and I have the same error if I put "http://subdomain.domain.name:80" or "http://subdomain.domain.name"

My domain name is recognized in public DNS and works for all of my other subdomain and I have valid server certificate. I have a HAProxy which do the SSL termination and redirect to Kalliope API in http.

**Comment by Sispheor** [comment]
The idea is to perform the SSL termination only to the frontend. Then a normal http call betwwen the front and the kalliope rest server.

**Comment by JulienSambre** [comment]
I agree and that's what I did. I do that way for alot of my applications (like Gotify for receiving Kalliope notifications on several phones). Protocol change from https to http after proxy is something that I manage everyday at work. I put a diagram of what I did to be sure :).

![Kalliope Proxy](https://user-images.githubusercontent.com/14327288/61795041-d1978500-ae22-11e9-917f-0fbc18f67838.png)

Therefore, I don't manage to make it work for https and the error part "for URL: null" made me ask myself what is wrong. Long story short, how the app know when he must do a https call (between the app and the front proxy)?

For better understanding, could you tell me what kind of URL is supported? Because from what I tested, only [domain/ip]:[port] works and so, how does the app know when to do a https call?

**Comment by Sispheor** [comment]
Please go on Gitter when you have time to discuss this

**Comment by Sispheor** [comment]
It's a bug. We've hard coded the protocol.
@LaMonF see [here](https://github.com/kalliope-project/kalliope-app/blob/master/kalliope/src/pages/settings/settings.service.ts#L38) :)
